db/db_impl/db_impl.cc

```diff
@@ -5164,6 +5175,8 @@ Status DestroyDB(const std::string& dbname, const Options& options,
 
 Status DBImpl::WriteOptionsFile(bool need_mutex_lock,
                                 bool need_enter_write_thread) {
+  options_mutex_.AssertHeld();
+
   WriteThread::Writer w;
```

Comment:
I think entering write thread to serialize writing out OPTIONS files is redundant now. That'll be great if we can eliminate it since it blocks writes for a potentially long time

Reply:
Great point. I didn't really understand what that was doing here, and I think it's better to address that in a follow-up PR